libraries/ms-drivers/src/ads1259_adc.c

```diff
@@ -121,3 +122,172 @@ StatusCode ads1259_get_conversion_data(Ads1259Storage *storage) {
+  };
+  store_register(MX_STORE_TYPE__ADS1259, funcs, &s_store, NULL);
+}
+
+// Used to determine length of time needed between convert command sent and data collection
+static const uint32_t s_conversion_time_ms_lookup[NUM_ADS1259_DATA_RATE] = {
+  [ADS1259_DATA_RATE_10] = 100, [ADS1259_DATA_RATE_17] = 61,   [ADS1259_DATA_RATE_50] = 21,
+  [ADS1259_DATA_RATE_60] = 17,  [ADS1259_DATA_RATE_400] = 3,   [ADS1259_DATA_RATE_1200] = 2,
+  [ADS1259_DATA_RATE_3600] = 1, [ADS1259_DATA_RATE_14400] = 1,
+};
+
+// Number of noise free bits for each sampling rate
+static const uint8_t s_num_usable_bits[NUM_ADS1259_DATA_RATE] = {
+  [ADS1259_DATA_RATE_10] = 21,   [ADS1259_DATA_RATE_17] = 21,    [ADS1259_DATA_RATE_50] = 20,
+  [ADS1259_DATA_RATE_60] = 20,   [ADS1259_DATA_RATE_400] = 19,   [ADS1259_DATA_RATE_1200] = 18,
+  [ADS1259_DATA_RATE_3600] = 17, [ADS1259_DATA_RATE_14400] = 16,
+};
+
+// tx spi command to ads1259
+static void prv_send_command(Ads1259Storage *storage, uint8_t command) {
+  uint8_t payload[] = { command };
+  spi_exchange(storage->spi_port, payload, 1, NULL, 0);
+}
+
+static StatusCode prv_configure_registers(Ads1259Storage *storage) {
+  uint8_t register_lookup[NUM_CONFIG_REGISTERS] = {
+    (ADS1259_SPI_TIMEOUT_ENABLE | ADS1259_INTERNAL_REF_BIAS_ENABLE),
+    (ADS1259_OUT_OF_RANGE_FLAG_ENABLE | ADS1259_CHECK_SUM_ENABLE),
+    (ADS1259_CONVERSION_CONTROL_MODE_PULSE | ADS1259_DATA_RATE_SPS),
+  };
+  // reset all register values to default
+  prv_send_command(storage, ADS1259_STOP_READ_DATA_CONTINUOUS);
+  prv_send_command(storage, ADS1259_RESET);
+  delay_us(100);  // Needs 8 fclk cycles before next command
+  prv_send_command(storage, ADS1259_STOP_READ_DATA_CONTINUOUS);
+  uint8_t payload[NUM_REGISTER_WRITE_COMM] = { (ADS1259_WRITE_REGISTER | ADS1259_ADDRESS_CONFIG0),
+                                               NUM_CONFIG_REGISTERS - 1, register_lookup[0],
+                                               register_lookup[1], register_lookup[2] };
+  spi_exchange(storage->spi_port, payload, NUM_REGISTER_WRITE_COMM, NULL, 0);
+  // sanity check that data was written correctly
+  uint8_t check_regs[] = { (ADS1259_READ_REGISTER | ADS1259_ADDRESS_CONFIG0),
+                           NUM_ADS1259_REGISTERS - 1 };
+  uint8_t reg_readback[NUM_ADS1259_REGISTERS];
+  spi_exchange(storage->spi_port, check_regs, sizeof(check_regs), reg_readback,
+               NUM_ADS1259_REGISTERS);
+  LOG_DEBUG("ads1259 post init register values:\n");
+  for (int i = 0; i < NUM_ADS1259_REGISTERS; i++) {
+    LOG_DEBUG("reg %d: 0x%x\n", i, reg_readback[i]);
+  }
+  return STATUS_CODE_OK;
+}
+
+// calculate check-sum based on page 29 of datasheet
+static Ads1259StatusCode prv_checksum(Ads1259Storage *storage) {
+  uint8_t sum = (uint8_t)(s_store.rx_data_lsb + s_store.rx_data_mid + s_store.rx_data_msb +
+                          ADS1259_CHECKSUM_OFFSET);
+  LOG_DEBUG("STORAGE CHK_SUM: %x\n", (uint8_t)s_store.chk_sum);
+  LOG_DEBUG("CALCULATED SUM %x\n", sum);
```

Comment:
should probably not have logs here, logic is that if we're emulating a ramp or other continuous function in python for the current this would print a lot, it's also not super useful info (not obviously actionable / interpretable to reader)